### PR TITLE
Fixes #957 Fluo scripts copy jars into fluo software dir.

### LINKED ISSUE #957: Fluo scripts copy jars into fluo software dir.
For some commands, the fluo script will copy jars from HDFS to the local machine.  Currently the fluo script copies jars into `$FLUO_HOME/lib/app/<app name>`.  This has at least two problems.  First, its possible that commands running concurrently for the same app could interfere with each other.  Second, it possible that the software directory is not writable. 

One possible fix is the following :
 * Setup `FLUO_TMP` env var in `fluo-env.sh`.  Could default to `/tmp`
 * In scripts copy jars into dir like `mktemp -d $FLUO_TMP/fluo-"$app"-XXXXXXXXX`
 * Execute command and after if finishes remove tmp dir

### PR DESCRIPTION
**Pull request by mptap**
- [x] - Setup FLUO_TMP env var which defaults to /tmp in fluo-env.sh.
- [x] - Modified the fluo script to use mktemp and FLUO_TMP everywhere FluoGetJars is run.
- [x] - Modified the fluo script to remove tmp dir after if finishes executing command.
- [x] - Ran both `mvn package` and `mvn verify` successfully.
- [x] - Tested this locally using fluo w/ Uno.

### CODE REVIEW

**Comment by ctubbsii** [comment]
Umm, oops. I accidentally checked a box and unchecked it. That's weird. I didn't know you could create a checklist like that in a GitHub issue. Very cool.

File: modules/distribution/src/main/scripts/fluo [hunk 1]
```diff
@@ -143,7 +143,10 @@ function setup_service {
     app=${BASH_REMATCH[1]}
     verify_app "$app"
     check_conn_props
-    app_lib=$lib/apps/$1
+    # create a temp dir to fetch application jars to
+    app_lib=$(mktemp -d "$FLUO_TMP"/fluo-"$app"-XXXXXXXXX) || die "fatal: unable to allocate a temporary directory"
+    # schedule removal of app_lib tmp dir when this script exits
+    trap "rm -rf '""$app_lib""'" EXIT HUP INT QUIT TERM
     mkdir -p "$app_lib"
```
Comment (mikewalch) [inline modules/distribution/src/main/scripts/fluo]: Is `mkdir` command necessary now that the directory is created by `mktemp`?

File: modules/distribution/src/main/scripts/fluo [hunk 2]
```diff
@@ -225,7 +228,10 @@ exec)
     app=$2
     verify_app "$app"
     check_conn_props
-    app_lib=$lib/apps/$app
+    # create a temp dir to fetch application jars to
+    app_lib=$(mktemp -d "$FLUO_TMP"/fluo-"$app"-XXXXXXXXX) || die "fatal: unable to allocate a temporary directory"
+    # schedule removal of app_lib tmp dir when this script exits
+    trap "rm -rf '""$app_lib""'" EXIT HUP INT QUIT TERM
     mkdir -p "$app_lib"
```
Comment (mikewalch) [inline modules/distribution/src/main/scripts/fluo]: Same as above

**Comment by mptap** [comment]
@keith-turner: Thanks, it was a great first issue! Look forward to contributing more. For sure, my twitter is @manjiritapaswi

**Comment by mptap** [comment]
@mikewalch: Done! PR: https://github.com/apache/fluo-website/pull/105

**Comment by keith-turner** [comment]
> Thanks, it was a great first issue! Look forward to contributing more.

Glad to hear it @mptap.  Let us know if you need assistance finding other issues or have any other questions.

**Pull request merged**